src/printer.js

```diff
@@ -181,6 +181,20 @@ function genericPrintNoParens(path, options, print, args) {
     case "EmptyStatement":
       return "";
     case "ExpressionStatement":
+      // Detect Flow-parsed directives
+      if (n.directive) {
+        return concat([
+          nodeStr(
+            {
```

Comment:
Nooo, please don't fake objects that look like ast nodes. Instead it's better to add a boolean flag on the function argument saying if it's a directive or not and deal with it accordingly.